src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSubscriber.java

```diff
@@ -169,12 +172,46 @@ private void handle(String consumeTag, Delivery delivery) {
         }
     }
 
+    private boolean resubscribe() {
+        LOGGER.info("Trying to resubscribe to queue.");
+
+        RabbitMQConfiguration configuration = this.connectionManager.get().getConfiguration();
+        int attemptsCount = 0;
+        int amountOfAttempts = configuration.getMaxRecoveryAttempts();
+        boolean successful = false;
+
+        while (attemptsCount < amountOfAttempts) {
+            try {
+                Thread.sleep(configuration.getMaxConnectionRecoveryTimeout());
+            } catch (InterruptedException | IllegalArgumentException e) {
+                LOGGER.error(e.getMessage());
+            }
+
+            try {
+                attemptsCount++;
+                consumerMonitor.set(null);
+                start();
+
+                CommonMetrics.setReadiness(true);
+                successful = true;
+                break;
+            } catch (Exception e) {
+                if(attemptsCount >= amountOfAttempts) {
+                    LOGGER.error("Failed to start listening when resubscribing after {} attempts.", amountOfAttempts);
+                }
+            }
+        }
+
+        return successful;
+    }
+
     private void canceled(String consumerTag) {
         LOGGER.warn("Consuming cancelled for: '{}'", consumerTag);
-        try {
-            close();
-        } catch (Exception e) {
-            LOGGER.error("Can not close subscriber with exchange name '{}' and queues '{}'", exchangeName.get(), subscribeTarget.get(), e);
+
+        CommonMetrics.setReadiness(false);
```

Comment:
We should register the specific monitor for this case instead of using the user monitor